include/circt/Dialect/FIRRTL/Passes.td

```diff
@@ -850,4 +850,24 @@ def Lint :
   let constructor = "circt::firrtl::createLintingPass()";
 }
 
+def SpecializeOption :
+    Pass<"firrtl-specialize-option", "firrtl::CircuitOp"> {
```

Comment:
This could be a `FModuleOp` pass by my reading of https://mlir.llvm.org/docs/PassManagement/#operation-pass. What does the proposed error handling look like in that case?

Reply:
We'd get a bazillion copies of the error message? I think it's clearer handle parallelism within the pass in this case.